modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Environment.scala

```diff
@@ -279,7 +282,24 @@ object Environment {
     if (preparedRemoteAdapters.nonEmpty)
       Clients
         .mkHttp(remoteAdapters.connectionTimeout, remoteAdapters.readTimeout, remoteAdapters.maxConnections, ec)
+        .map(enableRemoteAdapterMetrics(metrics))
         .map(c => (c.some, preparedRemoteAdapters))
     else Resource.pure[F, (Option[Http4sClient[F]], Map[(String, String), RemoteAdapter])]((None, preparedRemoteAdapters))
   }
+
+  def enableRemoteAdapterMetrics[F[_]: Sync](metrics: Metrics[F])(inner: Http4sClient[F]): Http4sClient[F] =
+    Http4sClient { req =>
+      for {
+        resp <- inner.run(req).attemptTap {
+                  case Left(_: TimeoutException) => Resource.eval(metrics.remoteAdaptersTimeoutCount)
+                  case Left(_) => Resource.eval(metrics.remoteAdaptersFailureCount)
+                  case Right(response) =>
+                    response.status.responseClass match {
+                      case Status.Successful => Resource.eval(metrics.remoteAdaptersSuccessCount)
+                      case Status.ServerError => Resource.eval(metrics.remoteAdaptersFailureCount)
+                      case _ => Resource.pure[F, Unit](())
```

Comment:
What type of response does this last case match?  Currently this response is not captured by metrics, so there is no visibility to the pipeline operator.

Reply:
Last one corresponds to all responses except 2xx & 5xx (1xx, 3xx and 4xx). I think I should include 4xx in `remoteAdaptersFailureCount` as well, as it is a failure just like 5xx. However 1xx and 3xx aren't failures semantically.

Reply:
There is an inconsistency between what I do here and what's done at HttpClient Sync instance in terms of status code interpretation, I'll align both places

Reply:
I just sent a new commit to align both places, now any response without 2xx status code will be adapter failure and reported in `remoteAdaptersFailureCount` and any response with 2xx status code will be reported in `remoteAdaptersSuccessCount` where all cases are covered and made visible to the pipeline operator

wdyt @istreeter ?